diff --git a/node_modules/@capacitor/geolocation/android/build.gradle b/node_modules/@capacitor/geolocation/android/build.gradle
index 35f4a12..7fdde86 100644
--- a/node_modules/@capacitor/geolocation/android/build.gradle
+++ b/node_modules/@capacitor/geolocation/android/build.gradle
@@ -54,8 +54,8 @@ android {
         abortOnError = false
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_21
-        targetCompatibility JavaVersion.VERSION_21
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
     }
     publishing {
         singleVariant("release")
@@ -92,3 +92,4 @@ dependencies {
     androidTestImplementation "androidx.test.ext:junit:$androidxJunitVersion"
     androidTestImplementation "androidx.test.espresso:espresso-core:$androidxEspressoCoreVersion"
 }
+
